spicelib/editor/asc_editor.py

```diff
@@ -549,36 +552,38 @@ def _get_text_space(self):
         return min_x, max_y + 24  # Setting the text in the bottom left corner of the canvas
 
     def add_library_paths(self, *paths):
-        """Adding paths for searching for symbols and libraries"""
-        self.lib_paths.extend(*paths)
+        """
+        *(Deprecated)* Use the class method `set_custom_library_paths()` instead.
+        
+        Adding paths for searching for symbols and libraries"""
+        self.set_custom_library_paths(*paths)
 
     def _lib_file_find(self, filename) -> Optional[str]:
-        file_found = search_file_in_containers(filename, *self.lib_paths)  # The library paths
-        if file_found is None:
-            file_found = search_file_in_containers(filename,
-                                                   os.path.split(self.asc_file_path)[0],
-                                                   os.path.curdir,  # The current script directory,
-                                                   os.path.split(self.asc_file_path)[0],
-                                                   # The directory where the script is located
-                                                   os.path.expanduser("~/AppData/Local/LTspice/lib/sub"),
-                                                   os.path.expanduser("~/Documents/LtspiceXVII/lib/sub"),
-                                                   os.path.expanduser("~/AppData/Local/Programs/ADI/LTspice/lib.zip"),
-                                                   )
+        # create list of directories to search, based on the simulator_lib_paths. Just add "/sub" to the path
+        my_lib_paths = [os.path.join(x, "sub") for x in self.simulator_lib_paths]
+        # find the file
+        file_found = search_file_in_containers(filename, 
+                                               *self.custom_lib_paths,
+                                               os.path.split(self.asc_file_path)[0],  # The directory where the file is located
+                                               os.path.curdir,  # The current script directory,
```

Comment:
Not sure what would be the best order here. I think we should follow the same approach as the spice simulator. LTSpice searches for libraries where the .asc is located first. Should we do the same ?

Reply:
yes, I will reorder.

Reply:
reordered